frame-metadata/src/v16.rs

```diff
@@ -738,3 +594,33 @@ impl IntoPortable for DeprecationInfo {
 		}
 	}
 }
+
+/// A wrapper around [`codec::Compact`] which is compact
+/// encoded and can be transparently serialized via serde.
+// Dev note: If we add Serializing to codec::Compact, this
+// can be removed without altering the SCALE encoding.
+#[derive(Clone, Copy, PartialEq, Eq, Encode, Debug)]
+#[cfg_attr(feature = "decode", derive(Decode))]
+#[cfg_attr(feature = "serde_full", derive(Serialize))]
+#[cfg_attr(feature = "serde_full", serde(from = "T"))]
+pub struct CompactSer<T>(codec::Compact<T>);
```

Comment:
It's a bit ugly doing this, but the change should be transparent if we add serializing support to `parity-scale-codec`, and even in this case, it allows a wider range of parity-scale-copdec versions to work anyway.

Reply:
nvm, I missed that parity-scale-codec's "serde" feature flag covered Compat too.